From c48dd71953d06fbca5286d4f7cf246f68c1cff5f Mon Sep 17 00:00:00 2001
From: Andrew Liu <shengping.liu@windriver.com>
Date: Wed, 4 Aug 2010 15:24:39 +0800
Subject: [PATCH 161/252] bman: allow pool declarations on non-control-plane.

It is from FSL vendor SDK 2.x.

This allows other nodes (eg. ethernet) to cross-reference
to the buffer-pool nodes in order to configure their
buffer-pool use. However, the control-plane will need
a declaration of the same pool in order to reserve it
against dynamic allocation for another purposes.

Signed-off-by: Geoff Thorpe <Geoff.Thorpe@freescale.com>
Integrated-by: Andrew Liu <shengping.liu@windriver.com>
---
 drivers/hwalloc/bman_driver.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/hwalloc/bman_driver.c b/drivers/hwalloc/bman_driver.c
index fe37813..fe2ee4a 100644
--- a/drivers/hwalloc/bman_driver.c
+++ b/drivers/hwalloc/bman_driver.c
@@ -1,4 +1,4 @@
-/* Copyright (c) 2008, 2009 Freescale Semiconductor, Inc.
+/* Copyright (c) 2008-2010 Freescale Semiconductor, Inc.
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -69,7 +69,7 @@ static int __bm_pool_add(u32 bpid, u32 *cfg, int triplets)
 {
 	u64 total = 0;
 	BUG_ON((bpid + 1) > POOL_MAX);
-	if (bman_depletion_get(&pools, bpid)) {
+	if (bman_have_ccsr() && bman_depletion_get(&pools, bpid)) {
 		pr_err("Duplicate pool for bpid %d\n", bpid);
 		return -EBUSY;
 	}
@@ -103,8 +103,10 @@ static int __bm_pool_add(u32 bpid, u32 *cfg, int triplets)
 		bman_free_pool(pobj);
 		cfg += 6;
 	}
-	bman_depletion_set(&pools, bpid);
-	num_pools++;
+	if (bman_have_ccsr()) {
+		bman_depletion_set(&pools, bpid);
+		num_pools++;
+	}
 	if (total)
 		pr_info("Bman: reserved bpid %d, seeded %lld items\n", bpid,
 			total);
-- 
1.6.5.2

